sdk/eidolon_ai_sdk/cpu/conversational_apu.py

```diff
@@ -178,6 +232,48 @@ def tool_event_stream():
             await self.memory_unit.storeMessages(call_context, [message])
         conversation.append(message)
 
+    async def process_audio_message(self, call_context: CallContext, message: UserMessageAudio):
+        if self.audio_unit is None:
+            raise ValueError("No audio unit available")
+        audio_data, metadata = await AgentOS.process_file_system.read_file(call_context.process_id, message.file.file_id)
+        mimetype = metadata.get("mimetype")
+        path = metadata.get("path") or metadata.get("filename") or ""
+        text = await self.audio_unit.speech_to_text(audio=audio_data, mime_type=mimetype)
+        return [UserMessageText(text=f"The following text as converted from the audio file {path}:\n{text}\n\n")]
+
+    async def process_image_message(self, call_context: CallContext, message: UserMessageImage):
```

Comment:
The logic here is wrong.

you will never hit elif self.image_unit is not None since get_capabilities will always return true in this case.

I assume that you want to be checking the image capabilities on the llm unit instead of self.

I believe the other file types have a similar bug.

Reply:
I'm still working on this part.  It will change